### ISSUE TITLE
`relationJoins` MySQL converts nested Decimal to float

### ISSUE THREAD

**Issue by michaelhays** [prisma/prisma#23233]
### Bug description

When using `relationLoadStrategy: 'join'` in MySQL, retrieving nested models with `Decimal` fields seem to convert decimal values to floats.

### How to reproduce

https://github.com/michaelhays/prisma-relationjoins-decimal

See [these lines](https://github.com/michaelhays/prisma-relationjoins-decimal/blob/08d51d99444b890242a8e4892a25f1162914849f/run.ts#L28-L29) for what gets logged.

```ts
console.log(queryResult) // 95993.57
console.log(joinResult) // 95993.57000000001
```

### Prisma information

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["relationJoins"]
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model FirstModel {
  id           Int           @id @default(autoincrement())
  secondModels SecondModel[]
}

model SecondModel {
  id           Int        @id @default(autoincrement())
  value        Decimal
  firstModelId Int
  firstModel   FirstModel @relation(fields: [firstModelId], references: [id])

  @@index([firstModelId])
}
```

```ts
const joinResult = (await prisma.firstModel.findFirstOrThrow({
  relationLoadStrategy: 'join',
  include: { secondModels: true },
})).secondModels[0].value
```

### Environment & setup

- OS: Linux, Fedora 39
- Database: MySQL
- Node.js version: v20.11.0

### Prisma Version

```
prisma                  : 5.10.2
@prisma/client          : 5.10.2
Computed binaryTarget   : rhel-openssl-3.0.x
Operating System        : linux
Architecture            : x64
Node.js                 : v20.11.0
Query Engine (Node-API) : libquery-engine 5a9203d0590c951969e85a7d07215503f4672eb9 (at node_modules/@prisma/engines/libquery_engine-rhel-openssl-3.0.x.so.node)
Schema Engine           : schema-engine-cli 5a9203d0590c951969e85a7d07215503f4672eb9 (at node_modules/@prisma/engines/schema-engine-rhel-openssl-3.0.x)
Schema Wasm             : @prisma/prisma-schema-wasm 5.10.0-34.5a9203d0590c951969e85a7d07215503f4672eb9
Default Engines Hash    : 5a9203d0590c951969e85a7d07215503f4672eb9
Studio                  : 0.499.0
Preview Features        : relationJoins
```

**Comment by planklmar** [comment]
We found the same issue using Postgres, when using relationLoadStrategy: 'join'.
With load strategy relationLoadStrategy: 'query' or deactivating the preview Feature, the values are correct again.

In the query using "relationJoins", I can see a JSONB_BUILD_OBJECT maybe this is causing a wrong conversion to number at some point

prisma               : 5.10.2
@prisma/client : 5.10.2
Postgres           :   12
PreviewFeature: "relationJoins"

**Comment by Weakky** [comment]
Hey folks, this issue should be fixed in the next release of Prisma. Thanks for reporting 🙏

**Comment by michaelhays** [comment]
Confirmed that this is fixed in 5.12.0! Thanks so much y'all :)
